shell.nix

```diff
@@ -15,13 +15,25 @@ let
     udis86 # available since NixOS 19.09
     openal
   ];
+  runtimeLibPath = lib.makeLibraryPath runtimeLibs;
 in
 (mkClangShell {
   name = "factor-shell-env";
-  LD_LIBRARY_PATH = "/run/opengl-driver/lib:${lib.makeLibraryPath runtimeLibs}" ;
+  LD_LIBRARY_PATH = "/run/opengl-driver/lib:${runtimeLibPath}" ;
   buildInputs = runtimeLibs ++ [
     # for building factor
     git
     curl
+    makeWrapper
   ];
+  shellHook = ''
+    wrapFactor () {
+    [ -n "$1" ] || { printf "Usage: wrapFactor <factor-root>" ; return; }
+    local root="$(realpath $1)"
+    local binary="''${root}/factor"
+    wrapProgram "$binary" --prefix LD_LIBRARY_PATH : ${runtimeLibPath} \
+      --argv0 factor
+    ln -sf "''${root}/factor.image" "''${root}/.factor-wrapped.image"
```

Comment:
What's this symlink for?

Reply:
As far as I know, the factor binary searches for an image with the same basename next to it. `wrapProgram` moves the original `$binary` to `.${binary}-wrapped` and establishes a wrapper with the name `$binary`. Thus, I expect the wrapped factor binary to search for a wrapped-named factor image.

I'm just supplying the intent of this code. I don't know if any of my assumptions are correct.

Reply:
Yes, that is the reason for the symlink.